src/Microsoft.ML.Data/DataLoadSave/Text/TextLoaderCursor.cs

```diff
@@ -464,6 +469,176 @@ public LineBatch GetBatch()
+                                ichCur++;
+                        }
+
+                        if(startsInsideQuoted || line[ichCur] == '"')
+                        {
+                            // Quoted Field Case
+
+                            if (!startsInsideQuoted)
+                                ichCur++;
+
+                            for (; ; ichCur++)
+                            {
+                                if (ichCur >= ichLim)
+                                    // We've reached the end of the line without finding the closing quote,
+                                    // so next line will start on this quoted field
+                                    return true;
+
+                                if (line[ichCur] == '"')
+                                {
+                                    if (++ichCur >= ichLim)
+                                        // Last character in line was the closing quote of the field
+                                        return false;
+
+                                    if (line[ichCur] == '"')
+                                        // 2 Double quotes means escaped quote
+                                        continue;
+
+                                    // If it wasn't an escaped quote, then this is supposed to be
+                                    // the closing quote of the field, and there should only be spaces remaining
+                                    // until the next separator.
+
+                                    if (!_sepsContainsSpace)
+                                    {
+                                        // Ignore leading spaces
+                                        while (ichCur < ichLim && line[ichCur] == ' ')
+                                            ichCur++;
+                                    }
+
+                                    // If there's anything else than spaces or the next separator,
+                                    // this will actually be a QuotingError on the parser, so we decide that this
+                                    // line contains a quoting error, and so it's not going to be considered a valid field
+                                    // and the rest of the line should be ignored.
+                                    if (ichCur >= ichLim || IsSep(line[ichCur]))
+                                        return false;
+
+                                    quotingError = true;
+                                    return false;
+                                }
+                            }
+                        }
+
+                        // Unquoted field case.
+                        // An unquoted field shouldn't contain new lines
+                        while(ichCur < ichLim && !IsSep(line[ichCur]))
+                        {
+                            ichCur++;
+                        }
+                        return false;
+                    }
```

Comment:
I've rewritten the logic in Multilinereader, in order to be more flexible in what we accept (following this comment: https://github.com/dotnet/machinelearning/pull/5125#discussion_r427000494 )

In principle, it's the same as in previous iterations of the PR: simply read every character in the read line, and decide if we should read more lines as part of the current row. This code mimics the one found in [TextLoader.Parser.HelperImpl.FetchNextField()](https://github.com/dotnet/machinelearning/blob/24ca4439ed36b47918eeb074afef02cafb0356bc/src/Microsoft.ML.Data/DataLoadSave/Text/TextLoaderParser.cs#L1149-L1198)

So now the behavior has changed for the following invalid rows, which are now accepted by Multilinereader without much problem (although the parser might throw or log an error when processing these). Notice that these cases are actually not accepted by the RFC and can be considered invalid formats, so we actually have liberty on how we handle this, and I think the way I've handled it will cause the least negative impact for the users:

1. We'll accept " inside unquoted fields, and treat those quotes as if they were any other character. This one used to cause the multilinereader to think it was opening a quoted field, and potentially read the rest of the file into this line. Now it knows it's not opening a quoted field.
```
Label,Size,Age
true,5.4",5
```

2. We'll accept characters between a closing quote in a quoted field and the next separator. This case is actually always going to cause a [QuotingError on the Parser when parsing the row](https://github.com/dotnet/machinelearning/blob/24ca4439ed36b47918eeb074afef02cafb0356bc/src/Microsoft.ML.Data/DataLoadSave/Text/TextLoaderParser.cs#L1187-L1190) which will make the parser to return an empty row instead (notice that quoting errors don't cause exceptions). Because of this, we also set quotingError as true in here in the Multilinereader, which means that the rest of the row after the invalid closing quote, will be ignored:
```
Label,Size,Age
true,"5."4,5 // this will be read as a single line, and then the parser will return it as empty
false,"2.3",8// this will be read correctly
```

3. A positive side effect of point number 2, including the quotingError mechanism, is that now it's much harder for the multilinereader to read a complete file when there's a quoting error. Now if there's a quoted field without closing quote, it will read it until the next quote, and will consider all the read lines as a single row (which will produce a QuotingError on the parser). And so the following lines will remain unaffected. Previous behavior for this particular case would actually have been to load the rest of the file as part of the first row:
```
// the first 3 rows will be read as single row, and later the parser will have a quoting error there
// after that all rows will be read correctly.
Label,Size,Age
true,"5.4,5 
true,2,4
false,"2.3",8
true,"3.5",67
...
```